Gems/Atom/Asset/Shader/Code/Source/Editor/CommonFiles/Preprocessor.cpp

```diff
@@ -207,34 +252,44 @@ namespace AZ
             const char* inputPath = fullPath.c_str();
             // let's create the equivalent of that expression but in dynamic form:
             //const char* argv[] = { processName, szInPath, "-C", "-+", "-D macro1"..., "-I path"..., NULL };
-            AZStd::vector< const char* > argv;
-            argv.reserve(5 + options.m_predefinedMacros.size() * 2 + options.m_projectIncludePaths.size() * 2);
+            AZStd::vector< AZStd::string > argv;
+            argv.reserve(2 + cppArguments.size() + projectIncludePaths.size() * 2);
```

Comment:
Comment about where the magic numbers come from would be nice

Reply:
Done.